addon/components/polaris-color-picker.js

```diff
@@ -63,97 +64,117 @@ export default Component.extend({
 
     const backgroundColor = `rgba(${red}, ${green}, ${blue}, ${alpha})`;
     return htmlSafe(`background-color: ${backgroundColor};`);
-  }).readOnly(),
+  }
 
   /**
    * @private
    */
-  draggerX: computed('color.saturation', 'pickerSize', function() {
+  @(computed('color.saturation', 'pickerSize').readOnly())
+  get draggerX() {
     const {
       color: { saturation },
       pickerSize,
     } = this.getProperties('color', 'pickerSize');
     return clamp(saturation * pickerSize, 0, pickerSize);
-  }).readOnly(),
+  }
 
   /**
    * @private
    */
-  draggerY: computed('color.brightness', 'pickerSize', function() {
+  @(computed('color.brightness', 'pickerSize').readOnly())
+  get draggerY() {
     const {
       color: { brightness },
       pickerSize,
     } = this.getProperties('color', 'pickerSize');
     return clamp(pickerSize - brightness * pickerSize, 0, pickerSize);
-  }).readOnly(),
+  }
+
+  @computed('contentId')
+  get element() {
```

Comment:
Curious what @vladucu's thoughts are about this naming, mostly because I'm conflicted: part of me feels like we shouldn't be overwriting Ember's internal API, part of me feels like we _should_ be able to overwrite because _that's why we do inheritance_. Basically I'm good with this as is but would also understand if we'd prefer to name it something else 😝

Reply:
hmm, tbh...I'm not sure I understand why we need this

Reply:
@vladucu you're confused as to why it's a CP, or confused as to why we need to set the element?

I guess we could just set it in the init hook since that's the only place it's used. I was just trying to keep the same API that we were used to (having the use of `this.element`) but I can get rid of it. If we ever end up needing it in multiple methods then we can refactor to a CP again.

Reply:
yeah, why we need the CP....we get the `element` explicitly as first parameter in the action called through `did-insert` modifier....we could just set it here as a property on the component....but unless I'm wrong, it's only used in this one place

Reply:
update - was going through other files, and components like `polaris-color-picker/slidable` call `this.element` from multiple places, so maybe it's easiest to have it as a CP in this particular case?

If not we can call `document.querySelector('#${this.contentId}')` each time. Pretty much the same thing IMO 🤷‍♂

Reply:
note: maybe I'm missing something and just talking non-sense...feel free to correct me
what I mean is....we can just keep it simple....why add a CP when you already have an action that set's `contentId` and then have the CP compute what you call `element` now

even more...could this  just be? Not sure where the `content` stuff comes from in this PR....
```js

/**
 * Polaris color picker component.
 * See https://polaris.shopify.com/components/forms/color-picker
 */
@tagName('')
@templateLayout(layout)
export default class PolarisColorPicker extends Component {
  ....

// NO LONGER NEEDED
//  didRender() {
//    super.didRender(...arguments);

//    let element = this.element;

//    if (isNone(element)) {
//      return;
//    }

    // Grab the size of the picker for positioning the draggable markers.
//    const mainColorElement = element.querySelector(
//      'div.Polaris-ColorPicker__MainColor'
//    );
//    if (isNone(mainColorElement)) {
//      return;
//    }

//    this.set('pickerSize', mainColorElement.clientWidth);
//  }

  @action
  setPickerSize(element) {
    // Grab the size of the picker for positioning the draggable markers.
    const mainColorElement = element.querySelector(
      'div.Polaris-ColorPicker__MainColor'
    );
    this.set('pickerSize', mainColorElement.clientWidth);
  }

  ...
}
```

```hbs
<div
  id={{this.contentId}}
  class="Polaris-ColorPicker"
  {{did-insert this.setContentId}}
>
  <div class="Polaris-ColorPicker__MainColor">
    ...
  </div>
  ....
</div>
```

Reply:
^ Agreed, this works for `polaris-color-picker`, but other components will have multiple methods that need to access the container element, so in other components we should at least set `element` as a property on `did-insert` (see `slidable.js`, it accessing `element` in other places besides the `did-insert` method)

Reply:
don't get me wrong Tom....in the same action you call from `did-insert`, of course, keep a reference to the `mainColorElement` or whatever you need by setting it on the component